### PR TITLE
fix #1813

### LINKED ISSUE #1813: Can't find functional component with shallowMount
### Subject of the issue

If I mount my component using `shallowMount`, I can't find functional components inside it.

### Steps to reproduce

https://github.com/Djaler/vue-test-utils-functional-components-shallow-mount
1. Clone repo and install dependencies (`npm install`)
2. Run tests (`npm run test`)

### Expected behaviour

Icon component should be found and both tests should pass

### Actual behaviour

Icon component not found in both cases

### PR DESCRIPTION
**Pull request by Djaler**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch.
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuejs/vue-test-utils/blob/dev/.github/CONTRIBUTING.md#development-setup
- [x] New/updated tests are included

### CODE REVIEW

File: packages/test-utils/src/matches.js [hunk 1]
```diff
@@ -56,13 +56,7 @@ export function matches(node, selector) {
...
   }
 
-  const isFunctionalSelector = isConstructor(selector.value)
-    ? selector.value.options.functional
-    : selector.value.functional
-
-  const componentInstance = isFunctionalSelector
-    ? node[FUNCTIONAL_OPTIONS]
-    : node.child
+  const componentInstance = node[FUNCTIONAL_OPTIONS] || node.child
```
Comment (lmiller1990) [inline packages/test-utils/src/matches.js:59]: Looks like the line you removed was for back compat. You might need to skip the failing test. Are you able to update the PR and fix the failing test?

**Review by lmiller1990 — CHANGES_REQUESTED** [review]

**Comment by lmiller1990** [comment]
Thanks for this!

File: packages/test-utils/src/matches.js (hunk 1 above)
Comment (Djaler) [reply packages/test-utils/src/matches.js:59]: Should I just skip this test for vue < 2.3?

Comment (Djaler) [reply packages/test-utils/src/matches.js:59]: Because I find no way to throw exception in this case

Comment (lmiller1990) [reply packages/test-utils/src/matches.js:59]: Sure, that's probably fine, go for it. I think most people will be on at least 2.5 by now.

**Pull request merged**